flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -107,7 +177,10 @@ else if (message instanceof JobClientMessages.SubmitJobAndWait) {
 						decorateMessage(new Status.Failure(new Exception(msg))),
 						ActorRef.noSender());
 
+				// cancel scheduler and inactivity triggered receive messages
 				getContext().unwatch(jobManager);
+				scheduler.cancel();
+				resetContextAndActor();
```

Comment:
Yes. Thanks for pointing that out. :)
I was earlier doing things a little differently. Forgot to fuse them.